Sources/Common/Background Queue/QueueStorage.swift

```diff
@@ -82,6 +94,8 @@ public class FileManagerQueueStorage: QueueStorage {
             return false
         }
 
+        self.inventory = inventory
+
         return fileStorage.save(type: .queueInventory, contents: data, fileId: nil)
```

Comment:
Are you referring to file deletion in other functions of `QueueStorage` such as `deleteExpired` or `delete`? 

Because those functions all use `getInventory()` and `saveInventory()`, the in-memory cache will be up-to-date after files are deleted. The only functions that should be accessing the `inventory` property is `getInventory()` and `setInventory()`. [I made a refactor PR](https://github.com/customerio/customerio-ios/pull/369) to enforce this rule of `inventory` should only be accessed by these 2 functions.

Reply:
Yes, it does and it looks much cleaner and fail-proof, thank you.